esinstall/src/rollup-plugins/rollup-plugin-wrap-install-targets.ts

```diff
@@ -27,12 +25,43 @@ export function rollupPluginWrapInstallTargets(
   logger: AbstractLogger,
 ): Plugin {
   const installTargetSummaries: {[loc: string]: InstallTarget} = {};
+  const cjsScannedNamedExports = new Map<string, string[]>();
+
+  /**
+   * Runtime analysis: High Fidelity, but not always successful.
+   * `require()` the CJS file inside of Node.js to load the package and detect it's runtime exports.
+   */
+  function cjsAutoDetectExportsRuntime(normalizedFileLoc: string): string[] | undefined {
+    try {
+      return Object.keys(require(normalizedFileLoc)).filter((imp) => imp !== 'default');
```

Comment:
This is very clever

Reply:
Do we need to detect if `require()` isn’t an object with keys first (e.g. `module.exports = myFunc()` or `module.exports = 'some string'`)? I guess that would fall into the `catch()` block but it may be a bit noisy

Reply:
ah, yea that's a good call. Yea that would be a "default export" anyway so the failure gives us the result we want, but yea I'd rather code defensively.